README.md

```diff
@@ -1,10 +1,170 @@
 
-## Adding a new repository
+The rust-vmm project is organized as a shared-effort, shared-ownership
+open-source project that includes (so far) contributors from Alibaba, AWS,
+Cloud Base, Crowdstrike, Intel, Google, Red Hat as well as individual
+contributors.
 
-1. Create the repository on your personal GitHub account.
-1. Submit a new issue [review request issue](https://github.com/rust-vmm/community/issues/new?assignees=&labels=&template=repository-review-request.md&title=Repository%2FCrate+Review+Request).
+Each virtualization component lives in a separated GitHub repository under the
+rust-vmm organization. The code is organized such that one repository
+corresponds to one
+[Rust crate](https://doc.rust-lang.org/book/ch07-01-packages-and-crates-for-making-libraries-and-executables.html).
+
+## Why rust-vmm
+
+- **Reduce code duplication**. The initial thought with rust-vmm was to create a
+  place for sharing common virtualization components between two existing VMMs
+  written in Rust:
+  [CrosVM](https://chromium.googlesource.com/chromiumos/platform/crosvm/) and
+  [Firecracker](https://github.com/firecracker-microvm/firecracker/). These
+  two projects have similar code for calling KVM ioctls, managing the
+  Virtual Machine memory, interacting with virtio devices and others. Instead
+  of having these components live in each of the projects repositories, they
+  can be shared through the rust-vmm project.
+- **Faster development**. rust-vmm provides a base of virtualization components
+  which are meant to be generic such that they can be consumed by other
+  projects besides CrosVM and Firecracker. One example that is often mentioned
+  is building a container specific VMM. By doing so, the container VMM can
+  reuse most of the rust-vmm components and just build the glue around these
+  components as well as an appropriate API for interacting with the container.
+- **Security & Testability**. Having independent components makes fuzz testing
+  easy to apply to each individual package. Our top priority is now
+  [vm-virtio](https://github.com/rust-vmm/vm-virtio) as it provides a
+  [virtio](https://www.oasis-open.org/committees/tc_home.php?wg_abbrev=virtio)
+  device implementation. We want to keep a high standard in terms of testing
+  as these virtualization packages are going to be used in production by
+  multiple projects. Each component is individually tested with a set of
+  common integration tests responsible for running unit tests and linters
+  (including coding style checks), and computing the coverage. In the future
+  we are planning to test the integration of the rust-vmm components by
+  providing a reference VMM implementation.
+- **Clean interface**. With shared crates you also have to think about an
+  interface flexible enough to be used by multiple VMMs. We have multiple
+  rounds of design reviews up to the point we are confident the interface is
+  clean and reusable by other projects.
+
+## rust-vmm crates
```

Comment:
So the whole sections below that one is a description of the crates statuses. I suggest we have 2 different sections:

* A description of the rust-vmm crates lifecycle: From initial crate request to then first PR merge and eventually making it to crates.io. And then also how code is getting pushed to crates.io. We should describe all of this in that section.

* An actual status of the current crates. I believe this could live in its own file, `crates.md` and be referenced here. The trick here is that either way it's often going to be outdated, and the only valid source for a given crate status is going to be the crate repo itself. So maybe we should not try to keep track of the crate statuses in a documentation file at all?

Reply:
> Regarding your first bullet, I believe the section Adding a new virtualization component should give the information you are talking about. Are you suggesting to move that section here? I thought it makes more sense in the development section but I can also move it.
> 
Partially, yes. It does not mention the different steps a crate is expected to follow, and how a crate can "graduate" from one step to the next one.


> Regarding the status, I think it is important to give a high level description of crates, we don't have to have it up to date every time and it is also impossible as you already state it. We can also have it in a different file, I don't mind, but I think it gives readers the opportunity to quickly scroll through all the existing components without having to go to every single repository. If someone is interested in an in-depth description of a particular component then I'll assume they'll go and read about it on the component repository.

Then maybe we should have it in a more concise form, a table for example 

| Crate  | Description | Status |
| ------------- | ------------- | ----------|
|  [kvm-ioctls](https://crates.io/crates/kvm-ioctls) |  Safe wrappers over the KVM API  |  Published on crates.io |
| [vm-memory](https://github.com/rust-vmm/vm-memory) | rust-vmm memory model | Merged on rust-vmm |

Reply:
I find tables hard to read in text mode, but ok, sounds good.

Reply:
I take that back. Tables are absolutely hideous in markdown so if you don't feel strongly against having it with sections instead of tables I would leave it as is. I'll add some more details about the status and how we get from one stage to the other.

Reply:
They're not nice in markdown, but much easier to read when rendered.